HTTP_API_Guide.md

```diff
@@ -0,0 +1,206 @@
+* accept: Content-Types. The server currently supports application/json. Future versions may support other types.
+* accept-encoding: Compression formats. The server supports gzip encoding (i.e. accept-encoding: gzip).
+* accept-version: HTTP API version. This document describes version 1.0.0. We encourage clients to explicitly specify this header; however, if it is not specified, the server defaults to 1.0.0.
+
+
+Response format
+---------------
+
+The HTTP response message body consists of a JSON object with the following
+properties:
+
+* status: If we successfully communicated with the Open API, this will be 0. Otherwise, it will contain details about the error.
+* message: If we successfully communicated with the Open API, this will be "OK". Otherwise, it will contain a description of the error.
+* data: The response from the Open API (only present if message = "OK"). The exact structure of this value depends on the operation, and is detailed in section A.2.14 of the [Developer's Guide].
+
+Note that message = "OK" doesn't necessarily mean that the request successfully
+got useful data. If the HTTP server successfully makes the Open API call but
+there was an error in executing the operation (e.g. because a required
+parameter for that operation was not specified), then status = 0 and
+message = "OK" but the data returned from the API will contain an error (e.g.
+in a responseError property of the data, or a securityError property of one or
+more of the securityData).
+
+Below is the response from the example request in the previous section. Notice
+that the response JSON at the top level has just the three properties, and that
+the data property is where all the interesting stuff is happening. For more
+details about the shape of HistoricalData responses, consult section A.2.5 in
+the [Developer's Guide].
+
+```
+{ "data":[
+    { "securityData":
+      { "security": "IBM US Equity",
+        "eidData":[],
+        "sequenceNumber":0,
+        "fieldExceptions":[],
+        "fieldData":[{"date":"2012-01-03T00:00:00.000Z","PX_LAST":186.3,"OPEN":186.73},{"date":"2012-01-04T00:00:00.000Z","PX_LAST":185.54,"OPEN":185.57},{"date":"2012-01-05T00:00:00.000Z","PX_LAST":184.66,"OPEN":184.81}]
+      }
+    },
+    { "securityData":
+      { "security": "AAPL US Equity",
+        "eidData":[],
+        "sequenceNumber":1,
+        "fieldExceptions":[],
+        "fieldData":[{"date":"2012-01-03T00:00:00.000Z","PX_LAST":58.7471,"OPEN":58.4857},{"date":"2012-01-04T00:00:00.000Z","PX_LAST":59.0629,"OPEN":58.5714},{"date":"2012-01-05T00:00:00.000Z","PX_LAST":59.7186,"OPEN":59.2786}]
+      }
+    }],
+  "status":0,
+  "message":"OK" }
+```
+
+
+Reference Data Service (blp/refdata)
+====================================
+
+The Reference Data Service provides request/response access to market data. It
+supports several kinds of requests, each with its own request parameters and
+response types. The sections below describe the more interesting/useful types
```

Comment:
The sub-sections below describe the different types of requests/responses that are available via the [HTTP API].